Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionSidebar.h

```diff
@@ -0,0 +1,142 @@
+ */
+- (void)sidebarWasEnabled;
+
+/*!
+ @abstract Called when the sidebar has changed from an enabled state to a disabled state.
+ */
+- (void)sidebarWasDisabled;
+
+/*!
+ @abstract Called when the sidebar's icon has been updated
+ */
+- (void)iconWasUpdated;
+
+@end
+
+/*!
+ @abstract A `_WKWebExtensionSidebar` object encapsulates the properties for a specific web extension sidebar.
+ @discussion When this property is `nil`, it indicates that the action is the default action and not associated with a specific tab.
+ */
+WK_CLASS_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA))
+NS_SWIFT_NAME(WKWebExtension.Sidebar)
+@interface _WKWebExtensionSidebar : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
+- (instancetype)new NS_UNAVAILABLE;
+
+/*! @abstract The extension context to which this sidebar is related. */
+@property (nonatomic, readonly, weak) WKWebExtensionContext *webExtensionContext;
+
+/*! @abstract The tab that this sidebar is associated with, or `nil` if it is the default sidebar */
+@property (nonatomic, readonly, nullable, weak) id <WKWebExtensionTab> associatedTab;
+
+/*! @abstract The title of this sidebar. */
+@property (nonatomic, readonly, copy) NSString *title;
+
+/*! @abstract Whether this sidebar is enabled or not. */
+@property (nonatomic, readonly, getter=isEnabled) BOOL enabled;
+
+/*! @abstract The web view which should be displayed when this sidebar is opened. */
+@property (nonatomic, readonly, nullable) WKWebView *webView;
+
+/*! @abstract The delegate for this sidebar which will receive callbacks when the title or web view is updated. */
+@property (nonatomic, nullable, weak) id <_WKWebExtensionActiveSidebarDelegate> activeSidebarDelegate;
+
+/*! @abstract The delegate for this sidebar which will receive callbacks when it is enabled, disabled, or the icon is changed. */
+@property (nonatomic, nullable, weak) id <_WKWebExtensionInactiveSidebarDelegate> inactiveSidebarDelegate;
+
+/*!
+ @abstract Indicate that the sidebar will be opened
+ @discussion This method should be invoked by the browser when this sidebar will be opened -- i.e., its associated ``WKWebView`` will be
+ displayed. If this method is not called before the sidebar is opened, then the ``WKWebView`` associated with this sidebar may not have a
+ document loaded.
+ */
+- (void)willOpenSidebar;
+
+/*!
+ @abstract Indicate that the sidebar will be closed
+ @discussion This method should be invoked by the browser when the sidebar will be closed -- i.e., its associated ``WKWebView`` will cease
+ to be displayed. If this method is not called when the sidebar is closed, then the sidebar's associated ``WKWebView`` may remain active longer than
+ necessary. Note that calling this method does not guarantee that the ``WKWebView`` associated with a particular sidebar will be deallocated, as the
+ web view may be shared between mutliple sidebars.
+ */
+- (void)willCloseSidebar;
```

Comment:
A view controller will help with this. But `WKWebExtensionAction` loads the web view on first access, and there is a `closePopup` method to close it. I would suggest similar here.